frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/GoService.java

```diff
@@ -0,0 +1,90 @@
+package ch.shiftcrypto.bitboxapp;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.Service;
+import android.content.Intent;
+import android.os.Binder;
+import android.os.Build;
+import android.os.IBinder;
+
+import androidx.core.app.NotificationCompat;
+
+import goserver.GoAPIInterface;
+import goserver.GoEnvironmentInterface;
+import goserver.Goserver;
+
+public class GoService extends Service {
+    // Binder given to clients
+    private final IBinder binder = new GoServiceBinder();
+
+    private Boolean started = false;
+
+    private final String channelId = "21";
+
+    private final int notificationId = 8;
+
+    @Override
+    public void onCreate() {
+        Util.log("GoService onCreate()");
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationChannel channel = new NotificationChannel(
+                    channelId,
+                    "BitBoxApp",
+                    NotificationManager.IMPORTANCE_LOW
+            );
+            channel.setDescription("BitBoxApp service notification channel");
+            NotificationManager manager = getSystemService(NotificationManager.class);
+            manager.createNotificationChannel(channel);
+        }
+
+        Notification notification = new NotificationCompat.Builder(this, channelId)
+                .setContentTitle("BitBoxApp Go backend service")
+                .setContentText("Running in the background")
+                .build();
+
+        // The service goes in foreground to keep working normally even when the app is out of
+        // focus. This is needed to avoid timeouts when the backend is polling the BitBox for e.g.
+        // an address verification.
+        startForeground(notificationId, notification);
+        Util.log("GoService onCreate completed");
+    }
+
+    @Override
+    public void onDestroy() {
+        Util.log("GoService onDestroy()");
+    }
+
+    public void startServer(String filePath, GoEnvironmentInterface goEnvironment, GoAPIInterface goAPI) {
```

Comment:
is there any chance this can be called concurrently? if so the `started` check is not thread-safe.

Reply:
Good point, I am not completely sure about that. I'll come back on this soon.

Reply:
I'm still not completely sure about this: it is generally correct to implement bound service's methods in a thread-safe way (see https://developer.android.com/guide/components/processes-and-threads), but in this case we have only one activity that actually binds to the service, so maybe we could avoid that.
It seems safer to handle it in a thread-safe way though: I added a lock guarding the `started` flag.